### PR TITLE
Philly-Hip Disaster Recovery(DR)

### PR DESCRIPTION
**Pull request by ronardcaktus**
Creates DR for the philly-hip web application

### CODE REVIEW

**Comment by copelco** [comment]
Task linked: [CU-30tprqy HIP Philly](https://app.clickup.com/t/30tprqy)

File: deploy/host_vars/dr.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+env_django_secret_key: !vault |
+  $ANSIBLE_VAULT;1.1;AES256
+  63323239653538366637613837333237633038383531396663616236663436363431306234623962
+  6230313632646239653933366636643537663762306630660a383135656336336362363364343765
+  38313838303233616437633462353965393563393937323332613333643461323963333831336536
+  6165306163363938320a653864333161326437613438616533383364386636373734613539336565
+  64386133396266636230663137383031383939396431623637643764353832386538386361666531
+  3238323337313164306432323331363636353437346335333435
+
+k8s_auth_api_key:
```
Comment (ronardcaktus) [inline deploy/host_vars/dr.yml:29]: How do I get the api key again?

File: deploy/group_vars/all.yml [hunk 2]
```diff
@@ -86,7 +86,7 @@ k8s_aws_fluent_bit_chart_version: "0.1.18"
 
 k8s_auth_host: "{{ ClusterEndpoint }}"
 k8s_auth_ssl_ca_cert: "k8s_auth_ssl_ca_cert.txt"
-k8s_namespace: "{{ app_name }}-{{ env_name }}"
+k8s_cluster_name: "{{ app_name }}-{{ env_name }}"
```
Comment (copelco) [inline deploy/group_vars/all.yml:89]:
```suggestion
k8s_namespace: "{{ app_name }}-{{ env_name }}"
```

I'm not sure I understand this one. `k8s_namespace` needs to remain. Perhaps `k8s_cluster_name` should also be added here as `philly-hip-stack-cluster`?

image

File: deploy/group_vars/all.yml [hunk 3]
```diff
@@ -95,6 +95,15 @@ k8s_container_port: 8000
...
 k8s_container_image_pull_policy: Always
 k8s_container_replicas: 2
+
+# Lower resources to preserve Node resources
+k8s_container_resources:
+    requests:
+        memory: "256Mi"
+        cpu: "50m"
+    limits:
+        cpu: "250m"
```
Comment (copelco) [inline deploy/group_vars/all.yml:105]:
```suggestion
```

File: deploy/group_vars/staging_shared.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# aws eks describe-cluster --name=philly-hip-stack-cluster | grep endpoint
+ClusterEndpoint: https://C3219F3CB49E4B1C82CFE8C82A846345.sk1.us-east-1.eks.amazonaws.com
+# aws rds describe-db-instances
+DatabaseAddress: pd13w6wwn2hbn7f.cp7c2yqiusbp.us-east-1.rds.amazonaws.com
+
```
Comment (copelco) [inline deploy/group_vars/staging_shared.yml:5]:
```suggestion
```

Move back to all.yml

File: deploy/group_vars/staging_shared.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+# ----------------------------------------------------------------------------
+# caktus.django-k8s
+# ----------------------------------------------------------------------------
+
+k8s_auth_ssl_ca_cert: staging_k8s_auth_ssl_ca_cert.txt # hard-coded env_name for sharing 
+k8s_cluster_name: hip-staging # hard-coded env_name for sharing 
+
+# aws eks describe-cluster --name=philly-hip-stack-cluster --query 'cluster.arn'
+k8s_context: "arn:aws:eks:us-east-1:061553509755:cluster/philly-hip-stack-cluster" 
+
```
Comment (copelco) [inline deploy/group_vars/staging_shared.yml:15]:
```suggestion
```

There's only one cluster, so these should all be in `all.yml`

File: deploy/group_vars/staging_shared.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+# Disaster Recovery: DB restore configuration
+k8s_restore_namespace: "{{ k8s_namespace }}"
+k8s_restore_target_db_url: "{{ env_database_url }}"
+k8s_restore_maint_user: hip-philly
+k8s_restore_maint_host: "{{ DatabaseAddress }}"
+k8s_restore_maint_port: "5432"
+k8s_restore_maint_name: hip-philly
+
+k8s_restore_maint_pass: # Need pass
```
Comment (copelco) [inline deploy/group_vars/staging_shared.yml:35]: https://github.com/caktus/philly-hip/blob/48feab7d5697b224bc2e1cd1e6a639bb1de54445/deploy/group_vars/all.yml#L21-L29

`ansible -m debug all -a "var= admin_database_password"`

File: deploy/group_vars/staging_shared.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  38653066646362323434626564333166653433663261643465356538386265353239336131633061
+  3236316135343961646363316531383661623336623564336637
+
+# Disaster Recovery: DB restore configuration
+k8s_restore_namespace: "{{ k8s_namespace }}"
+k8s_restore_target_db_url: "{{ env_database_url }}"
+k8s_restore_maint_user: hip-philly
+k8s_restore_maint_host: "{{ DatabaseAddress }}"
+k8s_restore_maint_port: "5432"
+k8s_restore_maint_name: hip-philly
```
Comment (copelco) [inline deploy/group_vars/staging_shared.yml:33]: https://github.com/caktus/philly-hip/blob/48feab7d5697b224bc2e1cd1e6a639bb1de54445/deploy/group_vars/all.yml#L60

```suggestion
k8s_restore_maint_name: hip
```

File: deploy/group_vars/staging_shared.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  3463373731653734393761303235626230623833316161660a666232376136663338303435623638
+  62623630383437396562393432313565316663363961396565656438336333393866313234663365
+  3830653833633431640a636333366134383934346632396263363430346539306436383031316436
+  38653066646362323434626564333166653433663261643465356538386265353239336131633061
+  3236316135343961646363316531383661623336623564336637
+
+# Disaster Recovery: DB restore configuration
+k8s_restore_namespace: "{{ k8s_namespace }}"
+k8s_restore_target_db_url: "{{ env_database_url }}"
+k8s_restore_maint_user: hip-philly
```
Comment (copelco) [inline deploy/group_vars/staging_shared.yml:30]: https://github.com/caktus/philly-hip/blob/48feab7d5697b224bc2e1cd1e6a639bb1de54445/deploy/group_vars/all.yml#L61

```suggestion
k8s_restore_maint_user: hip_admin
```

File: deploy/host_vars/staging.yml [hunk 9]
```diff
@@ -57,15 +57,6 @@ k8s_auth_api_key: !vault |
...
   363132353764303637656564323632636334
 
-database_password: !vault |
-  $ANSIBLE_VAULT;1.1;AES256
-  31633236353234623634616635306235633431373936393632656266333831333335326565353536
-  6565353434336561386234346639626634363164613139620a333134643231643634373137323638
-  62366363386436613935323163663562313865326236306662333765613565353037386332323134
-  3631346233636465350a646135386165336631383837383436653638346361333862663035613066
-  61383363666334303639353661353862363433383833643164623865636363383162343666636132
-  3135306661623561633630323062613065623738383866653833
```
Comment (copelco) [inline deploy/host_vars/staging.yml:67]: This value should be preserved and moved to `staging_shared.yml`, but I see a different encrypted password there

File: docs/backups.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+* **PostgreSQL database backups:** Backups are stored in the `hip-production-assets` (us-east-2) S3 bucket.
+* **Uploaded media backups:** S3 objects are replicated from `hip-production-philly-assets` (us-east-1) to `hip-dr-assets` (us-east-2).
+
+## Backup Verification Procedures
+
+Please follow the workflow outlined in [Disaster
+Recovery](https://caktus.github.io/developer-documentation/reference/disaster-recovery/).
+
+Additional documentation for backup verifications can be found here: [Backups: Kubernetes Backups](https://docs.google.com/document/d/16ke-22G1m04la-9X2kuR_QKSvXrnNXAx-pr5VTuBRgE/edit#)
+
```
Comment (copelco) [inline docs/backups.md:17]:
```suggestion
```

**Review by copelco — COMMENTED** [review]
Added some initial comments!

File: deploy/group_vars/staging_shared.yml (hunk 8 above)
Comment (ronardcaktus) [reply deploy/group_vars/staging_shared.yml:30]: Makes sense. App name and app_name + admin

**Comment by ronardcaktus** [comment]
## Backup Verification

Bucket `hip-production-philly-backups` shows a daily pg dump, latest dump was yesterday (March 22 2023)
Screenshot 2023-03-23 at 3 41 18 PM

`https://dr-hip.caktus-built.com/` shows a change of a month ago - see `Newsletters`
Screenshot 2023-03-23 at 3 43 17 PM


Successfully uploaded an image of Elly the 🐘 to `https://dr-hip.caktus-built.com/` 
Screenshot 2023-03-23 at 3 42 32 PM

**Pull request merged**